javaslang/src/main/java/javaslang/collection/TreeMap.java

```diff
@@ -457,6 +475,16 @@ public boolean containsKey(K key) {
     }
 
     @Override
+    public Option<Tuple2<K, V>> max() {
+        return entries.max();
+    }
+
+    @Override
+    public Option<Tuple2<K, V>> min() {
+        return entries.min();
```

Comment:
I've took a look at `head` impl. Looks fine, nothing to do here. These methods should use directly the backing structure `entries` (which is already done)